### PR TITLE
change: move the disable to _meta

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes #7482

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by soulbird** [comment]
Need update doc here: https://github.com/apache/apisix/blob/master/docs/en/latest/terminology/plugin.md#plugin-common-configuration

**Comment by tzssangglass** [comment]
> Need update doc here: [https://github.com/apache/apisix/blob/master/docs/en/latest/terminology/plugin.md#plugin-common-configuration](https://github.com/apache/apisix/blob/master/docs/en/latest/terminology/plugin.md?rgh-link-date=2022-08-17T06%3A31%3A11Z#plugin-common-configuration)

updated

**Review by tokers — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Review by starsz — APPROVED** [review]

File: apisix/plugin.lua [hunk 1]
```diff
@@ -761,8 +770,11 @@ local function check_single_plugin_schema(name, plugin_conf, schema_type, skip_d
     end
 
     if plugin_obj.check_schema then
-        local disable = plugin_conf.disable
-        plugin_conf.disable = nil
+        local disable = check_disable(plugin_conf)
+        if disable ~= nil then
+            plugin_conf._meta.disable = nil
```
Comment (spacewander) [inline apisix/plugin.lua:775]: Once we have moved the `disable` to _meta, we don't need to delete it during the check_schema. As there is no longer a top level extra field in the conf.

File: docs/en/latest/terminology/plugin.md [hunk 2]
```diff
@@ -80,10 +80,25 @@ Some common configurations can be applied to plugins through the `_meta` configu
...
 | Name         | Type | Description |
 |--------------|------|-------------|
+| disable      | boolean  | Whether to disable the plugin |
 | error_response | string/object  | Custom error response |
 | priority       | integer        | Custom plugin priority |
 | filter  | array | Depending on the requested parameters, it is decided at runtime whether the plugin should be executed. Something like this: `{{var, operator, val}, {var, operator, val}, ...}}`. For example: `{"arg_version", "==", "v2"}`, indicating that the current request parameter `version` is `v2`. The variables here are consistent with NGINX internal variables. For details on supported operators, please see [lua-resty-expr](https://github.com/api7/lua-resty-expr#operator-list). |
 
+### Disable the plugin
+
+Through the `disable` configuration, you can add a new plugin with disabled status and the request will not go through the plugin. This feature is used in the APISIX Dashboard.
```
Comment (spacewander) [inline docs/en/latest/terminology/plugin.md:90]: We don't need to mention the Dashboard as this feature can be used by any other components too.

Comment (tzssangglass) [reply docs/en/latest/terminology/plugin.md]: done

File: apisix/plugin.lua (hunk 1 above)
Comment (tzssangglass) [reply apisix/plugin.lua]: fixed

**Review by spacewander — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**